### PR TITLE
OCPBUGS-34607: Disable machine pool feature gate

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -3065,6 +3065,9 @@ func reconcileCAPIProviderDeployment(deployment *appsv1.Deployment, capiProvider
 	// Enforce pull policy.
 	for i := range deployment.Spec.Template.Spec.Containers {
 		deployment.Spec.Template.Spec.Containers[i].ImagePullPolicy = corev1.PullIfNotPresent
+		if deployment.Spec.Template.Spec.Containers[i].Name == "manager" {
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:3068]: why is this not done in CAPIProviderDeploymentSpec for azure?

Comment (bryan-cox) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go]: Discussed in Slack. Moved to include this flag in `CAPIProviderDeploymentSpec` for Azure.

**Pull request merged**